internal/overlord/logstate/syslog_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package logstate
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/canonical/pebble/internal/servicelog"
+	. "gopkg.in/check.v1"
+)
+
+type syslogSuite struct{}
+
+var _ = Suite(&syslogSuite{})
+
+func (s *syslogSuite) TestEncodeEntry(c *C) {
+	client := &syslogClient{
+		version:        1,
+		pid:            "100146",
+		host:           "mycontainer",
+		msgid:          "-",
+		priority:       priorityVal(FacilityUserLevelMessage, SeverityInformational),
+		structuredData: syslogStructuredData("pebble", canonicalPrivEnterpriseNum, nil),
+	}
+	buf := &bytes.Buffer{}
+	client.encodeEntry(buf, servicelog.Entry{
+		Time:    time.Date(2021, 5, 26, 12, 37, 1, 0, time.UTC),
+		Service: "foo",
+		Message: "this is a log entry",
+	})
+	c.Check(buf.String(), Equals, "73 <14>1 2021-05-26T12:37:01Z mycontainer foo 100146 - - this is a log entry")
+}
+
+func (s *syslogSuite) TestSyslogClient(c *C) {
+	// Start a fake syslog server
+	ln, err := net.Listen("tcp", "localhost:0") // will select unused port
+	c.Assert(err, IsNil)
+	defer ln.Close()
+
+	messages := make(chan string)
+	srv := &testSyslogServer{
+		listener: ln,
+		messages: messages,
+	}
+
+	go func() {
+		err := srv.run()
+		c.Assert(err, IsNil)
```

Comment:
I don't think we call `srv.close()`? Is there a potential race condition here, where the test might finish before `srv.run()` does and this assert is called? (And then `c` might be invalid as it has no where to report the test failure.) So I think we should do a "wait for closed channel with timeout" type of thing here as well.